.ci/bumpStackReleaseVersion.groovy

```diff
@@ -0,0 +1,137 @@
+  }
+  parameters {
+    booleanParam(name: 'DRY_RUN_MODE', defaultValue: false, description: 'If true, allows to execute this pipeline in dry run mode, without sending a PR.')
+  }
+  stages {
+    stage('Checkout') {
+      steps {
+        git(credentialsId: '2a9602aa-ab9f-4e52-baf3-b71ca88469c7-UserAndToken', url: "https://github.com/${ORG_NAME}/${REPO}.git")
+      }
+    }
+    stage('Fetch latest versions') {
+      steps {
+        script {
+          latestVersions = artifactsApi(action: 'latest-release-versions')
+        }
+        archiveArtifacts 'latest-release-versions.json'
+      }
+    }
+    stage('Send Pull Request'){
+      options {
+        warnError('Pull Requests failed')
+      }
+      steps {
+        generateSteps(projects: [ 'apm-pipeline-library' ])
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def generateSteps(Map args = [:]) {
+  args.projects.each { project ->
+    matrix( agent: 'linux && immutable',
+            axes:[
+              axis('REPO', [project.repo])
+            ]
+    ) {
+      bumpStackVersion(repo: env.REPO,
+                       scriptFile: '.ci/bump-release-version.sh',
+                       branch: 'master')
+    }
+  }
+}
+
+def bumpStackVersion(Map args = [:]){
+  catchError(buildResult: 'SUCCESS', stageResult: 'UNSTABLE') {
+    def arguments = prepareArguments(args)
+    createPullRequest(arguments)
+  }
+}
+
+def prepareArguments(Map args = [:]){
+  def repo = args.containsKey('repo') ? args.get('repo') : error('prepareArguments: repo argument is required')
+  def scriptFile = args.containsKey('scriptFile') ? args.get('scriptFile') : error('prepareArguments: scriptFile argument is required')
+  def branch = args.containsKey('branch') ? args.get('branch') : error('prepareArguments: branch argument is required')
+  def labels = args.get('labels', '').replaceAll('\\s','')
+  log(level: 'INFO', text: "prepareArguments(repo: ${repo}, branch: ${branch}, scriptFile: ${scriptFile}, reusePullRequest: ${reusePullRequest}, labels: '${labels}')")
+
+  def title = '[automation] update elastic stack release version'
+  def message = createPRDescription(latestVersions)
+  if (labels.trim() && !labels.contains('automation')) {
+    labels = "automation,${labels}"
+  }
+  return [repo: repo, branchName: branch, title: title, labels: labels, scriptFile: scriptFile, stackVersions: latestVersions, message: message]
+}
+
+def createPullRequest(Map args = [:]) {
+  prepareContext(repo: args.repo, branchName: args.branchName)
+  if (!args?.stackVersions?.trim()) {
+    error('createPullRequest: stackVersions is empty. Review the artifacts-api for the branch ' + args.branchName)
+  }
+  sh(script: """git checkout -b "update-stack-version-\$(date "+%Y%m%d%H%M%S")-${args.branchName}" """, label: "Git branch creations")
+  sh(script: "${args.scriptFile} '${args.stackVersions[0]}' '${args.stackVersions[1]}'", label: "Prepare changes for ${args.repo}")
```

Comment:
As far as I'm aware of the artifacts-api will only contain two releases:

![image](https://user-images.githubusercontent.com/2871786/116393562-b2a8fc80-a819-11eb-9d6f-aca0d9f69ccf.png)


The first element is the latest release, and the second one , aka last, is the upcoming release